content/en/docs/How Tos/metrics.md

```diff
@@ -134,13 +72,171 @@ or an error saying why it can't.
 func newMyMetrics(icfg interface{}) (baker.MetricsClient, error)
 ```
 
-#### Metrics.Client interface
+## How to publish statistics from a component
+
+All components need to implement a `Stats` method where they can expose metrics, called 
+by Baker once per second. Each of the different component types support a set of predefined
+metrics. In particular, the following counters are defined:
+
+- [Input](https://pkg.go.dev/github.com/AdRoll/baker#InputStats):
+    - `NumProcessedLines`, the total number of processed records since the component creation
+- [Filter](https://pkg.go.dev/github.com/AdRoll/baker#FilterStats):
+    - `NumProcessedLines`, the total number of processed records since the component creation
+    - `NumFilteredLines`, the number of filtered out (i.e. discarded) records
+- [Output](https://pkg.go.dev/github.com/AdRoll/baker#OutputStats):
+    - `NumProcessedLines`, the total number of processed records since the component creation
+    - `NumErrorLines`, the number of records that have produced an error
+- [Upload](https://pkg.go.dev/github.com/AdRoll/baker#UploadStats):
+    - `NumProcessedFiles`, the total number of processed files since the component creation
+    - `NumErrorFiles`, the number of files that have produced an error
+
+Due to historical reasons, these fields have the word _lines_ in them but they do 
+mean the number of records.
+
+The code example of 
+[how-to-create-filter](https://getbaker.io/docs/how-tos/create_filter/#processing-records) 
+shows how to correctly report metrics in a custom Filter.
+
+### Report custom metrics
+
+In addition to the records counter described above, components can report custom metrics
+giving a more specific view about the component health or performance.
+Baker supports two ways to expose custom metrics, namely:
+
+- return a 
+[`baker.MetricsBag`](https://pkg.go.dev/github.com/AdRoll/baker#MetricsBag)
+instance from the `Stats` method 
+- directly use the 
+[`baker.MetricsClient`](https://pkg.go.dev/github.com/AdRoll/baker#MetricsClient) in 
+the component code
+
+The two mechanisms follow the pull vs push approach, respectively.
```

Comment:
I guess we should use plural here since we're not speaking about a single "push-pull" approach, but 2 different approaches: a push approach and a pull one, so:
```suggestion
The two mechanisms follow the pull vs push approaches, respectively.
```